src/ThinkCoreBE.Api/Controllers/CustomerController.cs

```diff
@@ -17,5 +22,13 @@ private static async Task<IResult> GetAllCustomers(ICustomerService customerServ
             return Results.Ok(customers);
         }
 
+        private static async Task<IResult> DeleteCustomerById([FromQuery] long id, ICustomerService customerService, CancellationToken cancellationToken)
+        {
+            var deletedCount = await customerService.DeleteCustomerByIdAsync(id, cancellationToken);
+            if (deletedCount > 0) 
```

Comment:
não acho que esse if na camada de controller está ideal, por mais que ele seja pra retornar um httpResponse. Seria interessante ter ele na camada de serviço, e de alguma forma, abstrair esse resultado para poder definir se vamos retornar um OK, NotFound, etc.

GPT: By moving non-HTTP-related responsibilities to services or handlers, the controller can remain focused on its primary task: handling HTTP requests and returning HTTP responses. This will make your code more maintainable and in line with DDD best practices.

Reply:
Yes, creating a wrapper or middleware to handle common patterns like result wrapping and exception handling for all controllers is a great idea. This approach improves maintainability, reduces boilerplate, and ensures consistent behavior across your API. Here are some ways you can achieve this in a clean and reusable manner in a .NET DDD project:

Reply:
Option 4: Filters via IEndpointFilter